### Python Program to Retrieve Universities Data based on input Country Name

In [1]:
from get_universities_data import pipeline_universities

In [2]:
# list southeast asia countries
southeast_asia_countries = ["Brunei Darussalam", "Indonesia", "Cambodia", "Laos", "Malaysia", "Myanmar", "Philippines", "Singapore", "Thailand", "Vietnam"]

In [3]:
# GET DATA FROM API THEN SAVE TO JSON
southeast_asia_data = pipeline_universities(southeast_asia_countries, filename="southeast_asia_data")

INFO >> 1 - get universities data from 'Brunei Darussalam' (3)
INFO >> 2 - get universities data from 'Indonesia' (192)
INFO >> 3 - get universities data from 'Cambodia' (21)
INFO >> 4 - get universities data from 'Laos' (0)
INFO >> 5 - get universities data from 'Malaysia' (144)
INFO >> 6 - get universities data from 'Myanmar' (5)
INFO >> 7 - get universities data from 'Philippines' (118)
INFO >> 8 - get universities data from 'Singapore' (18)
INFO >> 9 - get universities data from 'Thailand' (67)
INFO >> 10 - get universities data from 'Vietnam' (2)
INFO >> Finished!
INFO >> DATA IS EXPORTEED TO d:\MyProject\AI-Engineer\ai-engineer-test\1_PythonSkill\southeast_asia_data.json


In [4]:
import pandas as pd

In [5]:
df = pd.DataFrame(southeast_asia_data)

In [6]:
df.head()

state-province alpha_two_code                                          name  \
0           None             BN                     Institut Teknologi Brunei   
1           None             BN               University of Brunei Darussalam   
2           None             BN            Universiti Islam Sultan Sharif Ali   
3           None             ID                        Universitas Nusa Putra   
4           None             ID  Akademi Farmasi Mitra Sehat Mandiri Sidoarjo   

                domains                         web_pages            country  
0          [itb.edu.bn]          [http://www.itb.edu.bn/]  Brunei Darussalam  
1          [ubd.edu.bn]          [http://www.ubd.edu.bn/]  Brunei Darussalam  
2       [unissa.edu.bn]       [http://www.unissa.edu.bn/]  Brunei Darussalam  
3     [nusaputra.ac.id]         [https://nusaputra.ac.id]          Indonesia  
4  [akfarmitseda.ac.id]  [http://www.akfarmitseda.ac.id/]          Indonesia

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 570 entries, 0 to 569
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   state-province  178 non-null    object
 1   alpha_two_code  570 non-null    object
 2   name            570 non-null    object
 3   domains         570 non-null    object
 4   web_pages       570 non-null    object
 5   country         570 non-null    object
dtypes: object(6)
memory usage: 26.8+ KB


In [8]:
# TOTAL UNIVERSITIES COUNTS PER COUNTRY NAME
df['country'].value_counts()

country
Indonesia            192
Malaysia             144
Philippines          118
Thailand              67
Cambodia              21
Singapore             18
Myanmar                5
Brunei Darussalam      3
Vietnam                2
Name: count, dtype: int64

Indonesia has the most number of universities (total=192) and Vietnam is the lowest (total=2).